specs/decoder.spec.ts

```diff
@@ -22,76 +22,117 @@ describe('Decoder', () => {
         }).toThrow(new Errors.AlgorithmNotSupported());
       });
     });
 
-    describe('verifying signature', () => {
-      const body = { foo: 'bar' };
+    describe("verifying signature", () => {
+      const body = { foo: "bar" };
       const token = generate(body);
 
-      it('verifies the signature successfully', () => {
+      it("verifies the signature successfully", () => {
         const actual = decoder.decodeAndVerify(token);
 
         expect(actual).toEqual(body);
       });
 
-      it('throws if signature is invalid', () => {
+      it("throws if signature is invalid", () => {
         expect(() => {
-          decoder.decodeAndVerify(token + 'XYZ');
+          decoder.decodeAndVerify(token + "XYZ");
         }).toThrow(new Errors.SignatureInvalid());
       });
 
-      it('validates claims', () => {
-        const spy = jest.spyOn(Verifier, 'verifyAll');
+      it("validates claims", () => {
+        const spy = jest.spyOn(Verifier, "verifyAll");
 
         decoder.decodeAndVerify(token);
 
         expect(spy).toHaveBeenCalledTimes(1);
       });
     });
 
-    describe('with generic types', () => {
+    describe("with generic types", () => {
       type TestBodyType = Record<string, number>;
 
       const typedDecoder = new Decoder<TestBodyType>(key);
       const body = { number: 42 };
       const token = generate(body);
 
-      it('accepts the generic type and returns the body', () => {
+      it("accepts the generic type and returns the body", () => {
         const actual = typedDecoder.decodeAndVerify(token);
 
         expect(actual).toEqual(body);
       });
     });
+
+    describe("with urlBase64 parse", () => {
+      type TestBodyType = Record<string, any>;
+      const typedDecoder = new Decoder<TestBodyType>(key);
+      const body = {
+        code: 1,
+        message: "OK",
+        data: {
+          user: {
+            level_experience_upper_limit: 9999,
+            total_level_experience: "xxx",
+            current_level_process_rate: 0.0996,
+            employer_invite_code: "BQ4W8Q",
+            level_reward_rate: 0.996,
+            id: "*****",
+            invite_code: "EQYUNC",
+            has_transactions: true,
+            level: 3,
+            avatar_url:
+              "https://app-xxxx.x.x-xx.xxx.com/public/development/avatarImage/924e97db-XXX-XX-ABAB-AA33ee9e3ecx.png?imageMogr2/crop/50x50",
+            first_name: "sss",
+            last_name: "ddd",
+            country: "japan",
+            balance: "*****",
+            invitees: 5,
+          },
+          user_token: "*****",
+        },
+      };
+      const token = generate(body);
+
+      it("Return the correct object containing URL-encoded data", () => {
+        const actual = typedDecoder.decodeAndVerify(token);
+
+        console.debug("🐛🐛🐛 ------------------------------🐛🐛🐛");
+        console.debug("🐛🐛🐛 ::: actual:::", actual);
+        console.debug("🐛🐛🐛 ------------------------------🐛🐛🐛");
+
```

Comment:
Could you please remove this debug logs?

Reply:
Done